atomic_reactor/util.py

```diff
@@ -738,21 +746,51 @@ def get_manifest_digests(image, registry, insecure=False, dockercfg_path=None,
 
         # Only compare prefix as response may use +prettyjws suffix
         # which is the case for signed manifest
-        response_h_prefix = response.headers['Content-Type'].rsplit('+', 1)[0]
-        request_h_prefix = media_type.rsplit('+', 1)[0]
-        if response_h_prefix != request_h_prefix:
-            logger.debug('request headers: %s', headers)
-            logger.debug('response headers: %s', response.headers)
-            logger.warning('Unable to fetch digest for %s', media_type)
-            continue
-
-        digests[version] = response.headers['Docker-Content-Digest']
-        context = '/'.join([x for x in [image.namespace, image.repo] if x])
-        tag = image.tag or 'latest'
-        logger.debug('Image %s:%s has %s manifest digest: %s',
-                     context, tag, version, digests[version])
-
-    if not digests:
+        try:
+            received_media_type = response.headers.get('Content-Type')
+        except KeyError:
+            # Guess content_type from contents
+            try:
+                encoding = guess_json_utf(response.content)
+                manifest = json.loads(response.content.decode(encoding))
+                received_media_type = manifest['mediaType']
+            except (ValueError,  # not valid JSON
+                    KeyError):   # no mediaType key
+                logger.warning("Unable to fetch media type: neither Content-Type header "
+                               "nor mediaType in output was found")
+
+        if received_media_type:
+            # set it to truthy value so that koji_import would know pulp supports these digests
+            digests[version] = True
+            response_h_prefix = received_media_type.rsplit('+', 1)[0]
+            request_h_prefix = media_type.rsplit('+', 1)[0]
+            if response_h_prefix != request_h_prefix:
+                logger.debug('request headers: %s', headers)
+                logger.debug('response headers: %s', response.headers)
+                logger.warning('Unable to fetch digest for %s', media_type)
+                continue
+
+        try
+            content_digest = response.headers['Docker-Content-Digest']:
+        except KeyError:
+            # Guess content_type from contents
+            try:
+                encoding = guess_json_utf(response.content)
+                manifest = json.loads(response.content.decode(encoding))
+                content_digest = manifest['config']['digest']
```

Comment:
Oh, okay. In that case its useless to use `get_manifest_digests` in pulp_pull - Pulp doesn't set this header